### PR TITLE
fix ipv6 address notation in urls

### PR DESCRIPTION
**Pull request by underdarknl**
This fixes the notation of  IPv6 addresses in the url fed to the requests lib.
Original traceback from the boefje:

```
Traceback (most recent call last):
  File "/var/www/html/.venv/lib/python3.8/site-packages/requests/models.py", line 434, in prepare_url
    scheme, auth, host, port, path, query, fragment = parse_url(url)
  File "/var/www/html/.venv/lib/python3.8/site-packages/urllib3/util/url.py", line 397, in parse_url
    return six.raise_from(LocationParseError(source_url), None)
  File "<string>", line 3, in raise_from
urllib3.exceptions.LocationParseError: Failed to parse: http://2a00:d00::10/

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/www/boefjes_v1.5.1/boefjes/job_handler.py", line 136, in handle
    boefje_results = self.job_runner.run(boefje_meta, environment)
  File "/var/www/boefjes_v1.5.1/boefjes/local.py", line 37, in run
    return boefje_resource.module.run(boefje_meta)
  File "/var/www/boefjes_v1.5.1/boefjes/plugins/kat_webpage_analysis/main.py", line 49, in run
    response = session.get(
  File "/var/www/html/.venv/lib/python3.8/site-packages/requests/sessions.py", line 600, in get
    return self.request("GET", url, **kwargs)
  File "/var/www/html/.venv/lib/python3.8/site-packages/requests/sessions.py", line 573, in request
    prep = self.prepare_request(req)
  File "/var/www/html/.venv/lib/python3.8/site-packages/requests/sessions.py", line 484, in prepare_request
    p.prepare(
  File "/var/www/html/.venv/lib/python3.8/site-packages/requests/models.py", line 368, in prepare
    self.prepare_url(url, params)
  File "/var/www/html/.venv/lib/python3.8/site-packages/requests/models.py", line 436, in prepare_url
    raise InvalidURL(*e.args)
requests.exceptions.InvalidURL: Failed to parse: http://2a00:d00::10/

Failed to parse: http://2a00:d00::10/
```

### CODE REVIEW

File: boefjes/plugins/kat_webpage_analysis/main.py [hunk 1]
```diff
@@ -33,7 +32,7 @@ def run(boefje_meta: BoefjeMeta) -> List[Tuple[set, Union[bytes, str]]]:
     else:
         # Fall back to old hack-ip-into-url behavior, for either https with no adapter, or http.
         if ip:
-            url_parts = url_parts._replace(netloc=ip)
+            url_parts = url_parts._replace(netloc="[%s]" % ip)
```
Comment (ammar92) [inline boefjes/plugins/kat_webpage_analysis/main.py:35]: Please prefer f"[{ip}]" over old style formats

**Review by ammar92 — APPROVED** [review]

File: boefjes/plugins/kat_webpage_analysis/main.py (hunk 1 above)
Comment (dekkers) [inline boefjes/plugins/kat_webpage_analysis/main.py:35]: As far as I know [] only work with IPv6, not with IPv4? At least wget won't accept it:

```
% wget 'http://[127.0.0.1]/'
http://[127.0.0.1]/: Invalid IPv6 numeric address.
```

So we probably need to check if the address is IPv4 or IPv6.

Comment (underdarknl) [reply boefjes/plugins/kat_webpage_analysis/main.py:35]: Requests has no problem with the notation with brackets for both ipv6 and ipv4.

Comment (ammar92) [reply boefjes/plugins/kat_webpage_analysis/main.py]: > Requests has no problem with the notation with brackets for both ipv6 and ipv4.

Are you sure about this? I tried to test this using requests 2.28.1 but I got a parse error for the bracketed IPv4 address but not for IPv6

Comment (noamblitz) [reply boefjes/plugins/kat_webpage_analysis/main.py]: Seems to work in KAT

**Comment by underdarknl** [comment]
We should add the ipaddress module to the requirements, and then I'd say this is ready to merging, right?

**Comment by ammar92** [comment]
`ipaddress` is not an external package. I think this one can be merged indeed, it has been updated and tested 👍

**Review by dekkers — APPROVED** [review]

**Pull request merged**